lib/hmmlearn/_hmmc.pyx

```diff
@@ -64,12 +67,78 @@ def _forward(int n_samples, int n_components,
 
                 fwdlattice[t, j] = _logsumexp(work_buffer) + framelogprob[t, j]
 
+def _forward_scaling(int n_samples, int n_components,
+                     dtype_t[:] startprob,
+                     dtype_t[:, :] transmat,
+                     dtype_t[:, :] frameprob,
+                     dtype_t[:, :] alpha,
+                     dtype_t[:] scaling_factors,
+                     dtype_t min_scaling=1e-10):
+    """
+    Compute the alpha probabilities using scaling_factors:
+        P(O_1, O_2, ..., O_t, q_t=S_i | model)
+    """
+    cdef:
+        Py_ssize_t t, i, j
+        dtype_t alpha_t
```

Comment:
I think what you call alpha here (consistently with the literature) is exp(fwdlattice) in the codebase?  let's stick to fwdlattice for now, we can always do a mass rename later (although I guess it may be a bit annoying that sometimes it includes the log and sometimes it doesn't).

Reply:
Fair Enough - Would be happy to help with making both algorithms match the naming in literature in the future.

Reply:
Yes, that would be great as a followup, but let's get this in first.